gcp/gcp.go

```diff
@@ -469,7 +470,7 @@ func marshalCapabilities(description *cdd.PrinterDescriptionSection) (string, er
 }
 
 // Share calls google.com/cloudprint/share to share a registered GCP printer.
-func (gcp *GoogleCloudPrint) Share(gcpID, shareScope string) error {
+func (gcp *GoogleCloudPrint) Share(gcpID, shareScope string, role string, skip_notification bool) error {
```

Comment:
Let's make role hard to misspell with something like this, then change type of `role` from `string` to `Role`:
```
type Role string
const (
  User Role = "USER"
  Manager Role = "MANAGER"
  Owner Role = "OWNER"
)
```